design/animation.md

```diff
@@ -0,0 +1,360 @@
+properties on a *particular* object, but instead reference a generic object of a given type by index. When bound,
+the animation will be given the list of objects which it will be modifying, and these path indices refer to that
+argument. Over the wire these are represented as strings, but are easily generated in the API with helper functions.
+
+```ts
+class TargetPath<T> {
+	private parent: TargetPath<any>;
+	constructor(name: string, children: Map<string, TargetPath<any>>) {
+		for (const [k,v] of children) {
+			this[k] = v;
+			v.parent = this;
+		}
+	}
+	public toJSON() {
+		// convert tree into string
+	}
+}
+function AnimateActor(name: string) {
+	return new TargetPath<never>(`actor.${name}`, {
+		transform: new TargetPath<never>("transform", {
+			local: new TargetPath<never>("local", {
+				position: new TargetPath<Vector3>("position", {
+					x: new TargetPath<number>("x"),
+					y: new TargetPath<number>("y"),
+					z: new TargetPath<number>("z")
+				})
+				// also rotation, scale
+			}),
+			app: new TargetPath<never>("app", {
+				/** lookAt replacement */
+				orientToward: new TargetPath<never>("orientToward", {
+					xy: new TargetPath<Vector3>("xy"),
+					y: new TargetPath<Vector3>("y")
+				})
+				// also position, rotation
+			})
+		})
+		// plus most other properties of actor
+	});
+}
+
+const track = {
+	target: AnimateActor("target").transform.local.position
+	// ...
+} as Track<Vector3>;
+```
+
+### References
+
+In addition to transitioning to a constant value on a property, you can also transition to a value on another object.
+This value is re-evaluated every frame on the client, so matches exactly.
+
+```ts
+assets.createAnimation({
+	targets: {
+		"start": "actor"
+		"end": "actor"
+	},
+	tracks: [{
+		target: AnimateActor("start").transform.local.position,
+		keyframes: [{
+			time: 1,
+			value: AnimateActor("end").transform.local.position
+		}]
+	}]
+});
+```
+
+### Instances
+
+Animation instances are animations that have been bound to a particular set of actors and materials. These instances
+can be created by explicitly binding an animation (`Animation.bind(...objects)`), or may be returned during actor
+creation from `Actor.CreateFromPrefab`/`Gltf`/`Library`. Unlike the asset, instances can be played, paused, set speed,
+etc. Animation instances can either be stored directly, or obtained from any of the actors/materials that participate
+in the animation. Instances participate in the patching system, so changing properties on the animation instance
+are pushed down to clients automatically.
+
+An animation instance can be set to start another animation on the instant that it finishes, by setting an instance's
+`continueWith` property. Because it's set in advance, there is zero downtime between one ending and another starting,
+and does not require a round trip.
+
+```ts
+class Animation {
+	bind(...args: Animatible[]): AnimationInstance;
```

Comment:
Does this method need to be updated to reflect the name-based binding? I like the name-based binding better than index-based.